File: BC-Android-Free/app/src/main/java/ca/josephroque/bowlingcompanion/dialog/ManualScoreDialog.java
package ca.josephroque.bowlingcompanion.dialog;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.InputFilter;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import ca.josephroque.bowlingcompanion.Constants;
import ca.josephroque.bowlingcompanion.R;

/**
 * Created by josephroque on 15-03-06.
 * <p/>
 * Location ca.josephroque.bowlingcompanion.dialog
 * in project Bowling Companion
 */
public class ManualScoreDialog extends DialogFragment
{

    private ManualScoreDialogListener mDialogListener;

    public static interface ManualScoreDialogListener
    {
        public void onSetScore(short gameScore);
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState)
    {
        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        final View dialogView = inflater.inflate(R.layout.dialog_set_score, null);

        final EditText editTextScore = (EditText)dialogView.findViewById(R.id.editText_score);
        editTextScore.setHint("Score (max 450)");
        editTextScore.setFilters(new InputFilter[]{new InputFilter.LengthFilter(3)});

        dialogBuilder.setView(dialogView)
                .setPositiveButton(Constants.DIALOG_OKAY, new DialogInterface.OnClickListener()
                {
                    @Override
                    public void onClick(DialogInterface dialog, int which)
                    {
                        short gameScore =
                                Short.parseShort(editTextScore.getText().toString());
                        mDialogListener.onSetScore(gameScore);
                        dialog.dismiss();
                    }
                })
                .setNegativeButton(Constants.DIALOG_CANCEL, new DialogInterface.OnClickListener()
                {
                    @Override
                    public void onClick(DialogInterface dialog, int which)
                    {
                        dialog.dismiss();
                    }
                });

        return dialogBuilder.create();
    }

    @Override
    public void onAttach(Activity activity)
    {
        super.onAttach(activity);

        try
        {
            mDialogListener = (ManualScoreDialogListener)activity;
        }
        catch (ClassCastException ex)
        {
            throw new RuntimeException("Parent activity must implement ManualScoreDialogListener");
        }
    }
}

File: legacy/app/src/main/java/ca/josephroque/bowlingcompanion/games/dialogs/ManualScoreDialog.kt
package ca.josephroque.bowlingcompanion.games.dialogs

import android.annotation.SuppressLint
import android.content.Context
import android.support.v7.app.AlertDialog
import ca.josephroque.bowlingcompanion.R
import android.view.LayoutInflater
import android.widget.EditText

/**
 * Copyright (C) 2018 Joseph Roque
 *
 * Present dialogs to set and clear a manual score in a game.
 */
object ManualScoreDialog {

    @SuppressLint("InflateParams")
    fun showSetScoreDialog(context: Context, onSetScore: (score: Int) -> Unit) {
        val view = LayoutInflater.from(context).inflate(R.layout.dialog_set_score, null, false)

        AlertDialog.Builder(context)
                .setTitle(R.string.dialog_set_score_title)
                .setView(view)
                .setPositiveButton(R.string.set_score) { dialog, _ ->
                    val scoreText = view.findViewById<EditText>(R.id.score)
                    if (scoreText.length() > 0) {
                        val gameScore = try {
                            scoreText.text.toString().toInt()
                        } catch (ex: NumberFormatException) {
                            -1
                        }

                        onSetScore(gameScore)
                    }
                    dialog.dismiss()
                }
                .setNegativeButton(R.string.cancel) { dialog, _ -> dialog.dismiss() }
                .create()
                .show()
    }

    fun showClearScoreDialog(context: Context, onClearScore: () -> Unit) {
        AlertDialog.Builder(context)
                .setTitle(R.string.dialog_clear_score_title)
                .setMessage(R.string.dialog_clear_score_message)
                .setPositiveButton(R.string.clear_score) { _, _ ->
                    onClearScore()
                }
                .setNegativeButton(R.string.cancel, null)
                .create()
                .show()
    }
}
